### PR TITLE
Add Collapsible Component

### PR DESCRIPTION
**Pull request by skattavar**
[Story](https://www.pivotaltracker.com/story/show/148396475)

This pull request replaces old pull request https://github.com/rentpath/react-ui/pull/25 which is closed.

### CODE REVIEW

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          <Button className={theme.strong} onClick={this.handleClick}>
+            {this.state.display ? visibleText : hiddenText}
+          </Button>
+        </div>
+      </div>
+    )
+  }
+}
+
+Collapsible.defaultProps = {
```
Comment (tcwilson) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: can you move `propTypes` and `defaultProps` up, into the object?  [See this for example](https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Button/Button.js#L11). The style you have here is fine if you're writing a functional component, but we prefer `static propTypes` and `static defaultProps` when extending `Component` or `PureComponent`.

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  showableItems: '',
+  nonshowableItems: '',
+  id: '',
+  title: '',
+  hiddenText: 'show more',
+  visibleText: 'show less',
+  visible: false,
+  handleClick: noop,
+}
```
Comment (tcwilson) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: you probably want to add a `theme` to `propTypes` and set the default prop of `theme` to `{}`. I.e.:

```javascript
static propTypes = {
  ...
  theme: PropTypes.object,
}

static defaultProps = {
  ...
  theme: {},
}
```

**Comment by tcwilson** [comment]
There are no changes to `package.json`, so the `yarn.lock` files should not be part of this PR. Can you remove these files? If they reappear when you run `yarn`, you may try removing your `node_modules` folders.

**Comment by tcwilson** [comment]
👍

File: packages/react-ui-core/src/Collapsible/__tests__/Collapsible-test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const { wrapper } = setup({ theme })
+    expect(wrapper.find('div').length).toBe(3)
+  })
+  it('contains show more', () => {
+    const { wrapper } = setup({ theme })
+    expect(wrapper.find('button').text()).toContain('more')
+  })
+  it('expands on click and contains show less', () => {
+    const { wrapper } = setup({ theme })
+    wrapper.find('button').simulate('click')
```
Comment (lfender6445) [inline packages/react-ui-core/src/Collapsible/__tests__/Collapsible-test.js]: are we missing a test case for the second click?

**Comment by lfender6445** [comment]
other than my comment I am 👍 please feel free to merge after

File: packages/react-ui-core/src/Collapsible/__tests__/Collapsible-test.js (hunk 3 above)
Comment (skattavar) [inline packages/react-ui-core/src/Collapsible/__tests__/Collapsible-test.js]: Added second click case as well.

File: packages/react-ui-core/src/Collapsible/Collapsible.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      id,
+      theme,
+      showableItems,
+      nonshowableItems,
+      hiddenText,
+      visibleText,
+    } = this.props
+
+    return (
+      <div id={id} className={theme.collapsible}>
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Collapsible/Collapsible.js]: This `id` should _not_ be here!

File: stories/examples/Collapsible.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { CollapsibleTheme } from '../theme'
+import { Collapsible } from 'react-ui-core/src/Collapsible'
+
+export default class Checkbox extends React.Component {
+  render() {
+  	return (<div className={this.props.className}>
```
Comment (tadjohnston) [inline stories/examples/Collapsible.js]: Ugh... this indentation should have been fixed before merge.

**Pull request merged**